dashboard/src/pages/ingest/batches/index.vue

```diff
@@ -0,0 +1,491 @@
+<script setup lang="ts">
```

Comment:
There's some leaking of store and domain details into this component; it'd be cool if it were more purely responsible for wiring the UI to the route and the stores, only expressing intentions like "apply these filters" or "go to page 2", instead of knowing how filters are represented internally or when query params like page need to be added or removed.